File: client/grao-a-grao/src/types/store.ts
export interface StoreModel {
    id: number;
    name: string;
    created_at: string;
    updated_at: string;
}

export interface CreateStoreRequest {
    name: string;
}

export interface UpdateStoreRequest {
    store_id: number;
    name: string;
}

export interface StoreResponse {
    id: number;
    name: string;
    created_at: string;
    updated_at: string;
}

// Normalize StoreResponse → StoreModel
export function normalizeStoreResponse(res: StoreResponse): StoreModel {
    return {
      id: res.id,
      name: res.name,
      created_at: res.created_at,
      updated_at: res.updated_at,
    };
  }
  
  // Convert StoreModel → UpdateStoreRequest
  export function toUpdateStoreRequest(model: StoreModel): UpdateStoreRequest {
    return {
      store_id: model.id,
      name: model.name,
    };
  }

File: pkg/dto/mapper/store.go
package mapper

import (
	"github.com/IlfGauhnith/GraoAGrao/pkg/dto/request"
	"github.com/IlfGauhnith/GraoAGrao/pkg/dto/response"
	"github.com/IlfGauhnith/GraoAGrao/pkg/model"
)

func CreateStoreToModel(req *request.CreateStoreRequest, userID uint) *model.Store {
	return &model.Store{
		Name:      req.Name,
		CreatedBy: model.User{ID: userID},
	}
}

func UpdateStoreToModel(req *request.UpdateStoreRequest, userID uint) *model.Store {
	return &model.Store{
		ID:        req.ID,
		Name:      req.Name,
		CreatedBy: model.User{ID: userID},
	}
}

func ToStoreResponse(m *model.Store) response.StoreResponse {
	return response.StoreResponse{
		ID:        m.ID,
		Name:      m.Name,
		CreatedAt: m.CreatedAt,
		UpdatedAt: m.UpdatedAt,
	}
}
